frontend/js/src/playlists/UserSearch.tsx

```diff
@@ -0,0 +1,96 @@
+import * as React from "react";
+import { faUser } from "@fortawesome/free-solid-svg-icons";
+import { throttle as _throttle } from "lodash";
+import debounceAsync from "debounce-async";
+import GlobalAppContext from "../utils/GlobalAppContext";
+import ListenControl from "../listens/ListenControl";
+
+export type UserSearchProps = {
+  userClick: (event: string) => void;
+  placeholder: string;
+};
+
+export type UserSearchState = {
+  newUser: string;
+  userSearchResults: Array<SearchUser>;
+};
+
+export default class UserSearch extends React.Component<
+  UserSearchProps,
+  UserSearchState
+> {
+  static contextType = GlobalAppContext;
+  declare context: React.ContextType<typeof GlobalAppContext>;
+
+  constructor(props: UserSearchProps) {
+    super(props);
+    this.state = {
+      newUser: "",
+      userSearchResults: [],
+    };
+  }
+
+  handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState(
+      {
+        newUser: event.target.value,
+      },
+      () => {
+        this.searchUsers();
+      }
+    );
+  };
+
+  // eslint-disable-next-line react/sort-comp
+  throttledHandleInputChange = _throttle(this.handleInputChange, 300);
```

Comment:
I think similarly to PR #2331 I gave you wrong instructions and this will not work well if you type quickly.
Instead of throttling `handleInputChange` we need to throttle the searchUsers and use `throttledSearchUsers` as the callback to setState in `handleInputChange`